### PR TITLE
pythonPackages.google-auth-oauthlib: init at 0.2.0

### LINKED ISSUE #57022: Missing package google-auth-oauthlib
## Issue description
Google has the following instructions to interact with G Suite, but it seems that we are missing `google-auth-oauthlib` in nixpkgs. 

> pip install --upgrade google-api-python-client google-auth-httplib2 google-auth-oauthlib

## Technical details

18.09/master

### PR DESCRIPTION
**Pull request by terlar**
###### Motivation for this change

I had this in my overlays and since there are other people using it I thought I would contribute it here.

fixes #57022 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/google-auth-oauthlib/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  checkPhase = ''
+    py.test
+  '';
+
+  meta = {
+    description = "Google Authentication Library: oauthlib integration";
+    homepage = https://github.com/GoogleCloudPlatform/google-auth-library-python-oauthlib;
+    license = lib.licenses.asl20;
+  };
```
Comment (dotlambda) [inline pkgs/development/python-modules/google-auth-oauthlib/default.nix]: Don't you want to maintain this?

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -2696,6 +2696,7 @@ in {
   googleapis_common_protos = callPackage ../development/python-modules/googleapis_common_protos { };
 
   google-auth-httplib2 = callPackage ../development/python-modules/google-auth-httplib2 { };
+  google-auth-oauthlib = callPackage ../development/python-modules/google-auth-oauthlib { };
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: Please add a blank line.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.google-auth-oauthlib python3.pkgs.google-auth-oauthlib

File: pkgs/development/python-modules/google-auth-oauthlib/default.nix (hunk 1 above)
Comment (terlar) [reply pkgs/development/python-modules/google-auth-oauthlib/default.nix]: I added myself as maintainer

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (terlar) [reply pkgs/top-level/python-packages.nix]: Sorry, should have figured from looking at all the other ones.

File: pkgs/development/python-modules/google-auth-oauthlib/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/google-auth-oauthlib/default.nix]: You should do so in a separate commit.

Comment (terlar) [reply pkgs/development/python-modules/google-auth-oauthlib/default.nix]: Okay, I thought about that, but wasn't sure about the convention to name such commits. I will check the log.

Comment (terlar) [reply pkgs/development/python-modules/google-auth-oauthlib/default.nix]: Done, sorry for the inconvenience, and thanks for the patience and keeping things tidy.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.google-auth-oauthlib python3.pkgs.google-auth-oauthlib

**Pull request merged**